src/middleware/FileRestrictionsMiddleware.ts

```diff
@@ -0,0 +1,92 @@
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys';
+import { UserProfileKeys } from 'ch-node-session-handler/lib/session/keys/UserProfileKeys';
+import { ISignInInfo, IUserProfile } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { NextFunction, Request, Response } from 'express';
+import { FORBIDDEN } from 'http-status-codes';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+import { loggerInstance } from './Logger';
+
+import { Appeal } from 'app/models/Appeal';
+import { AppealsPermissionKeys } from 'app/models/AppealPermissionKeys';
+import { APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { Attachment } from 'app/models/Attachment';
+
+const customErrorTemplate = 'error-custom';
+
+@provide(FileRestrictionsMiddleware)
+export class FileRestrictionsMiddleware extends BaseMiddleware {
+
+    public handler(req: Request, res: Response, next: NextFunction): void {
+
+        const session = req.session;
+
+        const adminPermissionFlag = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.AdminPermissions])
+            .extract();
+
+        const userProfile: IUserProfile = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.UserProfile])
+            .ifNothing(() => loggerInstance().error(`${FileRestrictionsMiddleware.name} - User profile was expected in session but none found`))
+            .unsafeCoerce();
+
+        const permissions = userProfile[UserProfileKeys.Permissions];
+
+        if (adminPermissionFlag === '1') {
+            if (permissions &&
+                permissions[AppealsPermissionKeys.download] &&
```

Comment:
can the permissions exist but be set to '0'? should we check the value is '1'?

Reply:
From what I observed the permissions are either there with a 1  if have them, or not at all. Extensions only check if they exist as well